crates/adapters/src/integrated/delta_table/output.rs

```diff
@@ -0,0 +1,515 @@
+                            }
+                        }
+                    } else {
+                        error!(
+                            "delta_table {}: received Data without a matching BatchStart",
+                            &inner.endpoint_name
+                        );
+                    }
+                }
+                None => {
+                    trace!(
+                        "delta_table {}: endpoint is shutting down",
+                        &inner.endpoint_name
+                    );
+                    return;
+                }
+            }
+        }
+    }
+}
+
+impl OutputConsumer for DeltaTableWriter {
+    fn max_buffer_size_bytes(&self) -> usize {
+        usize::MAX
+    }
+
+    fn batch_start(&mut self, _step: Step) {
+        self.command(Command::BatchStart)
+            .unwrap_or_else(|(e, fatal)| {
+                if let Some(controller) = self.inner.controller.upgrade() {
+                    controller.output_transport_error(
+                        self.inner.endpoint_id,
+                        &self.inner.endpoint_name,
+                        fatal,
+                        e,
+                    )
+                };
+            });
+    }
+
+    fn push_buffer(&mut self, _buffer: &[u8], _num_records: usize) {
+        unreachable!()
+    }
+
+    fn push_key(&mut self, _key: &[u8], _val: &[u8], _num_records: usize) {
+        unreachable!()
+    }
+
+    fn batch_end(&mut self) {
+        self.command(Command::BatchEnd)
+            .unwrap_or_else(|(e, fatal)| {
+                if let Some(controller) = self.inner.controller.upgrade() {
+                    controller.output_transport_error(
+                        self.inner.endpoint_id,
+                        &self.inner.endpoint_name,
+                        fatal,
+                        e,
+                    )
+                };
+            });
+    }
+}
+
+impl Encoder for DeltaTableWriter {
+    fn consumer(&mut self) -> &mut dyn OutputConsumer {
+        self
+    }
+
+    fn encode(&mut self, batch: &dyn SerBatchReader) -> AnyResult<()> {
+        let fields = self.inner.serde_arrow_schema.to_arrow_fields()?;
+        let mut builder = ArrowBuilder::new(&fields)?;
+
+        let mut num_records = 0;
+        let mut cursor = CursorWithPolarity::new(
+            batch.cursor(RecordFormat::Parquet(self.inner.serde_arrow_schema.clone()))?,
+        );
+        while cursor.key_valid() {
+            if !cursor.val_valid() {
+                cursor.step_key();
+                continue;
+            }
+            let mut w = cursor.weight();
+            if !(-MAX_DUPLICATES..=MAX_DUPLICATES).contains(&w) {
+                bail!("Unable to output record with very large weight {w}. Consider adjusting your SQL queries to avoid duplicate output records, e.g., using 'SELECT DISTINCT'.");
+            }
+            if w < 0 {
+                // TODO: we don't support deletes in the parquet format yet.
```

Comment:
bleah

Reply:
This should be coming soon with proper delta table support.